### PR TITLE
Use async refresh in get/mget

### PR DESCRIPTION
**Pull request by pxsalehi**
Follow-up to https://github.com/elastic/elasticsearch/pull/95869.

Relates ES-6047

### CODE REVIEW

**Review by Tim-Brooks — APPROVED** [review]
LGTM

**Comment by pxsalehi** [comment]
@elasticmachine update branch

File: server/src/main/java/org/elasticsearch/action/get/TransportShardMultiGetAction.java [hunk 1]
```diff
@@ -155,4 +150,21 @@ protected String getExecutor(MultiGetShardRequest request, ShardId shardId) {
             return super.getExecutor(request, shardId);
         }
     }
+
+    private void asyncShardMultiGet(MultiGetShardRequest request, ShardId shardId, ActionListener<MultiGetShardResponse> listener)
```
Comment (tlrx) [inline server/src/main/java/org/elasticsearch/action/get/TransportShardMultiGetAction.java:154]:
```suggestion
    private void asyncMultiGet(MultiGetShardRequest request, ShardId shardId, ActionListener<MultiGetShardResponse> listener)
```

Just to mirror the `asyncGet`

**Review by tlrx — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/action/get/TransportShardMultiGetAction.java (hunk 1 above)
Comment (pxsalehi) [reply server/src/main/java/org/elasticsearch/action/get/TransportShardMultiGetAction.java:154]: My intention was to use the same name as the action itself, hence the "shard" in the name. `TransportMultiGetAction` is slightly different and also exists (and gets broken down into `TransportShardMultiGetAction`).

Comment (tlrx) [reply server/src/main/java/org/elasticsearch/action/get/TransportShardMultiGetAction.java:154]: Ok

**Pull request merged**